agent/src/protocols/zeroconf/discovery_handler.rs

```diff
@@ -27,6 +25,35 @@ const DEVICE_HOST: &str = "AKRI_ZEROCONF_DEVICE_HOST";
 const DEVICE_ADDR: &str = "AKRI_ZEROCONF_DEVICE_ADDR";
 const DEVICE_PORT: &str = "AKRI_ZEROCONF_DEVICE_PORT";
 
+fn filter(config: &ZeroconfDiscoveryHandlerConfig, service: &ServiceDiscovery) -> bool {
+    trace!("[zeroconf:filter] Service: {:?}", service);
+    let include = (if let Some(name) = &config.name {
+        let result = name == service.name();
+        trace!("[zeroconf:filter] Name ({}) [{}]", name, result);
+        result
+    } else {
+        true
+    }) && (if let Some(domain) = &config.domain {
```

Comment:
i'm not familiar with zeroconf, is it possible to filter on name and domain?  if so, you might consider restructuring this.